lib/constructs/aws-serverless-s3-stack.ts

```diff
@@ -0,0 +1,115 @@
+import { NestedStack, NestedStackProps } from "aws-cdk-lib";
+import { Construct } from "constructs";
+import * as cdk from "aws-cdk-lib";
+import * as s3 from "aws-cdk-lib/aws-s3";
+import { AwsServerlessS3StackProps } from "./AwsServerlessS3StackProps";
+
+/**
+ * Represents a nested stack for creating and managing S3 buckets in an AWS Serverless application.
+ *
+ * This class creates three S3 buckets: a master bucket, a PDF files bucket, and a JSON files bucket.
+ * It also applies lifecycle rules to each bucket for efficient management of objects.
+ *
+ * @extends NestedStack
+ */
+export class AwsServerlessS3Stack extends NestedStack {
+    /** The name of the master S3 bucket */
+    public masterS3BucketName: string;
+    /** The name of the S3 bucket for storing PDF files */
+    public pfdFilesBucketName: string;
+    /** The name of the S3 bucket for storing JSON files */
+    public jsonFilesBucketName: string;
+
+    /**
+     * Constructs a new instance of the AwsServerlessS3Stack.
+     *
+     * @param {Construct} scope - The scope in which to define this construct.
+     * @param {string} id - The scoped construct ID.
+     * @param {AwsServerlessS3StackProps} props - Properties for configuring the stack.
+     * @throws {Error} If any required properties are missing from props.
+     */
+    constructor(scope: Construct, id: string, props: AwsServerlessS3StackProps) {
+        super(scope, id, props);
+
+        // create master S3 bucket
+        const masterS3Bucket = this.createS3Bucket(this, `${props.resourcePrefix}-${props.masterBucketName}`, cdk.RemovalPolicy.DESTROY);
+
+        // apply life cycle rules for master bucket
+        this.createS3BucketLifecycleRule(masterS3Bucket, `${props.resourcePrefix}-${props.masterBucketName}-LifecycleRule`);
+
+        // assign the master bucket as an input for the next stack
+        this.masterS3BucketName = masterS3Bucket.bucketName;
+
+        // create pdf files S3 bucket
+        const pfdFilesBucket = this.createS3Bucket(this, `${props.resourcePrefix}-${props.pfdFilesBucketName}`, cdk.RemovalPolicy.DESTROY);
+
+        // apply life cycle rules for pdf files bucket
+        this.createS3BucketLifecycleRule(pfdFilesBucket, `${props.resourcePrefix}-${props.pfdFilesBucketName}-LifecycleRule`);
+
+        // assign the pdf files bucket as an input for the next stack
+        this.pfdFilesBucketName = pfdFilesBucket.bucketName;
+
+        // create json files S3 bucket
+        const jsonFilesBucket = this.createS3Bucket(this, `${props.resourcePrefix}-${props.jsonFilesBucketName}`, cdk.RemovalPolicy.DESTROY);
```

Comment:
**Suggestion:** Add a check to ensure `props.resourcePrefix` is defined before using it to construct bucket names to avoid potential runtime errors.
> 🌖  **High Priority** 
 > [Category: **`Possible Issue`**, Importance: **`8`**]
> **Reason:** This suggestion prevents potential runtime errors by ensuring that `props.resourcePrefix` is defined before it is used, improving the robustness of the code.

```suggestion
        if (!props.resourcePrefix) {
          throw new Error('Resource prefix is missing in props.');
        }

        const masterS3Bucket = this.createS3Bucket(this, `${props.resourcePrefix}-${props.masterBucketName}`, cdk.RemovalPolicy.DESTROY);

        const pfdFilesBucket = this.createS3Bucket(this, `${props.resourcePrefix}-${props.pfdFilesBucketName}`, cdk.RemovalPolicy.DESTROY);

        const jsonFilesBucket = this.createS3Bucket(this, `${props.resourcePrefix}-${props.jsonFilesBucketName}`, cdk.RemovalPolicy.DESTROY);
```
- [ ]  **I have reviewed this code suggestion and addressed the relevant ones.**